### PR TITLE
logic / movies list based on api data rendering

### PR DESCRIPTION
**Pull request by traczoskar**
### What I did:

1.  I deleted redux-saga-effects dependency which was not useful and has caused errors. 
2. I installed framer-motion, very popular animation pack. I have used it for Tile components to animate a little fadeIn when they are getting visible in the window during scrolling. Let me know what you think. We can use it also for some else subtle animations. 
3. I added selector in Redux slice to select data status, changed some naming and changed initial status of movies for empty Array of Objects which responds to structure of API data.
4. I created new folder "utils" and the file "apiDataURLs.js" where I think we should store API addresses for crucial data, for example poster path, or backdrop which is separate URL. Then we can combine those URLs to create exact ones we want. 
5. I also moved getMoviesURL to utils folder.
6. In "MovieList" I  created switch which decides what to render which depends of status of the reducer.
7. I also added a camera Icon which appears when movie doesn't have a poster_path
8. I think we should consider creating a separate file "processingAPIdata.js" in utils where we gonna formatt data from API before injecting them into props. As you can see almost every data should be processed before we can use it in Tiles etc. For example Year of movie  comes as "2024-01-24" but it should be just "2024". So we won't do it directly in the MovieList component like I did now.
9. There's much more complicated with the genre tags. They're probably downloaded as a separately data and they're recognized by ID number. We should add separate downloading action, and I think we should store those genres in the localStorage of user. For now the genres are the same for all movies.
10. Some minor styling.

Hope you'll like it. If I forgot something, please check my commits.

Zrzut ekranu 2024-03-13 o 13 37 21

### CODE REVIEW

File: src/features/movies/MovieList/moviesListSlice.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { createSlice } from "@reduxjs/toolkit";
 
 const moviesSlice = createSlice({
-  name: "movies",
+  name: "moviesList",
   initialState: {
-    movies: null,
+    movies: [{}],
```
Comment (OlgaCybulska) [inline src/features/movies/MovieList/moviesListSlice.js:6]: Good idea, I wanted to change it

File: src/features/movies/MovieList/index.js [hunk 2]
```diff
@@ -6,34 +6,46 @@ import { SectionHeader } from "../../../common/SectionHeader";
...
+                <li key={movie.id}>
+                  <Tile
+                    posterPath={movie.poster_path}
+                    title={movie.original_title}
+                    year={movie.release_date.slice(0, 4)}
+                    rate={movie.vote_average.toFixed(1)}
+                    votes={movie.vote_count}
+                  />
+                </li>
+              ))}
```
Comment (OlgaCybulska) [inline src/features/movies/MovieList/index.js:45]: You can add pagination here.  when there are movies, it will be visible

**Review by OlgaCybulska — COMMENTED** [review]
good job :) just pagination and we can merge

File: src/features/movies/MovieList/getMovies.js
Comment (wojdaSoftware) [inline src/features/movies/MovieList/getMovies.js:1]: I think the additional error checks should occur here. Maybe check whether the response has a movie property and whether this property isn't empty?

File: src/common/LoadingPage/index.js
Comment (wojdaSoftware) [inline src/common/LoadingPage/index.js:1]: You removed the section wrapper. I don't have a problem with this, but this caused margin's to differ from the figma template. Right now the loading screen title is 40px away from the site header, while the movie list title is 56px away (as in the template). This causes a small but visible jump when the movies load in.

**Review by wojdaSoftware — CHANGES_REQUESTED** [review]
Error handling currently doesn't work properly. For example, if you make the request with the wrong API key the app tries to map an empty object. This happens, because the response is okay, but doesn't return the movies. It returns this payload instead:
image

So if our API key expires for some reason, the whole app crashes. I think we should have a more thorough error checking process.

There's also a small styling error I've described in one of the comments

Everything else is in order, the fade-in animation while scrolling is subtle, but looks great - good choice!

**Review by wojdaSoftware — APPROVED** [review]
good job!

**Pull request merged**